packages/core/src/types/index.ts

```diff
@@ -0,0 +1,32 @@
+export enum WorldEntryPoints {
+    get = "get",
+    set = "set",
+    entities = "entities",
+    execute = "execute"
+}
+
+export interface StorageKey {
+    partition: string,
+    keys: string[]
+}
+
+// TODO: extend Provider to this
+export interface IWorld {
+    constructor(executor_: string, store_: string, indexer_: string): void;
+    register_component(class_hash: string): void;
+    component(name: number): string;
```

Comment:
```suggestion
    component(name: number): string;
```

we'll need to represent felt252 as a BigInt since number can't express it